ml_git/plugin_interface/plugin_especialization.py

```
@@ -0,0 +1,31 @@
+"""
+© Copyright 2020 HP Development Company, L.P.
+SPDX-License-Identifier: GPL-2.0-only
+"""
+
+import importlib
+
+
+class PluginCaller:
+    SPEC_PLUGIN_KEY = 'data-plugin'
+
+    def __init__(self, manifest):
+        self.plugin_name = manifest.get(self.SPEC_PLUGIN_KEY, '')
+        self.package = self.__import_plugin()
+
+    def __import_plugin(self):
+        if not self.plugin_name:
+            return None
+        try:
+            package = importlib.import_module(self.plugin_name)
+            return package
+        except ImportError:
+            return None
```

If the user passed an invalid plugin and we detected it, shouldn't we report this error? I understand that if he passed the parameter in .spec it is because he wants to use it, so if ml-git is unable to import the package, it should be informed to the user.

---

I agree, ml-git must inform the error to the user.

---

I agree, ml-git must inform the error to the user.

---

Applied suggestions in commit 1c49752.